src/agent/src/main.rs

```diff
@@ -432,29 +438,183 @@ fn setup_debug_console(shells: Vec<String>, port: u32) -> Result<()> {
         let addr = SockAddr::new_vsock(libc::VMADDR_CID_ANY, port);
         socket::bind(listenfd, &addr)?;
         socket::listen(listenfd, 1)?;
-        socket::accept4(listenfd, SockFlag::SOCK_CLOEXEC)?
+        loop {
```

Comment:
This is going to break agent shutdown (see https://github.com/kata-containers/kata-containers/pull/411, specifically the calls to `select!()` on https://github.com/kata-containers/kata-containers/pull/411/commits/4465190f49e01a0675e03eee8b6d52eb891ed022).

Reply:
Thank you, @jodh-intel . That's mean we should hold this PR after #441 has been merged, and then adopt to the gracefully shutdown mechanism?